src/repo.rs

```diff
@@ -147,6 +159,32 @@ impl Repo {
         }
     }
 
+    /// List all repo objects
+    pub fn list_objects<P: IsA<gio::Cancellable>>(
+        &self,
+        flags: OstreeRepoListObjectsFlags,
+        cancellable: Option<&P>,
+    ) -> Result<HashSet<ObjectName>, Error> {
```

Comment:
I think the resulting entity here isn't a `HashSet<ObjectName>`. My understanding is that this would be an `HashMap` where the values are https://ostreedev.github.io/ostree/reference/ostree-OstreeRepo.html#OSTREE-REPO-LIST-OBJECTS-VARIANT-TYPE:CAPS.

Reply:
Would `HashMap<ObjectName, (bool, Vec<String>) ` as result-type meet your expectations?

Reply:
Yes, something like that. Even better, that tuple could be turned into its own `ObjectDetails` struct.